R/run-nmtran.R

```diff
@@ -0,0 +1,360 @@
+#' Interface for running `NM-TRAN` on model objects
+#'
+#' Function to run `NM-TRAN` on a model object to validate its control stream
+#'  for correct coding before submission. The `NM-TRAN` dataset (`FDATA`) and
+#'  other `NONMEM` artifacts can be further inspected by keeping the run directory
+#'  around.
+#'
+#' @details
+#' `NM-TRAN` is a preprocessor for `NONMEM` that translates user-specified
+#' control stream data and instructions into a form executable by `NONMEM`.
+#'
+#' Note that `nmtran_presort` is run ahead of `NM-TRAN` for `NONMEM 7.4` and later
+#'  - `nmtran_presort` is a supplementary utility that preprocesses the control
+#'     stream to ensure it is in the correct format for `NM-TRAN`.
+#'
+#' @param .mod A `bbi_nonmem_model` object.
+#' @param .bbi_args A named list specifying arguments to pass to `NM-TRAN`.
+#'   Similar to the `.bbi_args` argument defined in [submit_model()], though here
+#'   only `prdefault`, `tprdefault`, and `maxlim` arguments are passed to `NM-TRAN`.
+#'   `nm_version` is also supported and specifies which `NM-TRAN` executable to use.
+#'   See [print_bbi_args()] for more details.
+#' @inheritParams submit_model
+#' @param clean Logical (`T`/`F`). If `FALSE`, don't delete the temporary directory
+#'   containing the `NM-TRAN` run.
+#' @param run_dir Directory to run `NM-TRAN` in. Only relevant if `clean = FALSE`.
+#'
+#' @examples
+#' \dontrun{
+#'
+#' mod <- read_model(file.path(MODEL_DIR, 1))
+#' run_nmtran(mod, .bbi_args = list(nm_version = "nm74gf"))
+#'
+#' # Save the run directory for manual inspection
+#' run_nmtran(mod, clean = FALSE, run_dir = getwd())
+#'
+#' }
+#'
+#' @return An S3 object of class `nmtran_process`
+#' @export
+run_nmtran <- function(
+    .mod,
+    .bbi_args = NULL,
+    .config_path = NULL,
+    run_dir = tempdir(),
+    clean = TRUE
+){
+  check_model_object(.mod, "bbi_nonmem_model")
+
+  # Capture NONMEM/NM-TRAN options and format `NM-TRAN` args
+  nmtran_specs <- nmtran_setup(
+    .mod, .bbi_args = .bbi_args, .config_path = .config_path
+  )
+
+  mod_path <- get_model_path(.mod)
+  data_path <- get_data_path_from_ctl(.mod)
+
+  # NM-TRAN run directory
+  temp_folder <- withr::local_tempdir(
+    pattern = paste0("nmtran-mod_", get_model_id(.mod), "-"),
+    tmpdir = run_dir, clean = clean
+  )
+
+  # Copy model
+  fs::file_copy(mod_path, temp_folder)
+  nmtran_mod <- new_model(file.path(temp_folder, basename(mod_path)))
+
+  # Copy dataset & overwrite $DATA record of new model
+  # NM-TRAN will error if data cannot be found
+  if(fs::file_exists(data_path)){
+    new_data_path <- file.path(temp_folder, "data.csv")
```

Comment:
follow-up to <https://github.com/metrumresearchgroup/bbr/pull/705#discussion_r1693612429>

It probably makes sense to have a code comment explaining why "data.csv" is chosen rather than reusing the original data name.